interface/ceed-operator.c

```diff
@@ -1248,6 +1248,79 @@ int CeedOperatorGetNumQuadraturePoints(CeedOperator op, CeedInt *num_qpts) {
   return CEED_ERROR_SUCCESS;
 }
 
+/**
+  @brief Estimate number of FLOPs required to apply CeedOperator on the active vector
+
+  @param op    Operator to estimate FLOPs for
+  @param flops Address of variable to hold FLOPs estimate
+
+  @ref Backend
+**/
+int CeedOperatorGetFlopsEstimate(CeedOperator op, CeedInt *flops) {
+  int ierr;
+  bool is_composite;
+  ierr = CeedOperatorCheckReady(op); CeedChk(ierr);
+
+  *flops = 0;
+  ierr = CeedOperatorIsComposite(op, &is_composite); CeedChk(ierr);
+  if (is_composite) {
+    CeedInt num_suboperators;
+    ierr = CeedOperatorGetNumSub(op, &num_suboperators); CeedChk(ierr);
+    CeedOperator *sub_operators;
+    ierr = CeedOperatorGetSubList(op, &sub_operators); CeedChk(ierr);
+
+    // FLOPs for each suboperator
+    for (CeedInt i = 0; i < num_suboperators; i++) {
+      CeedInt suboperator_flops;
+      ierr = CeedOperatorGetFlopsEstimate(sub_operators[i], &suboperator_flops);
+      CeedChk(ierr);
+      *flops += suboperator_flops;
+    }
+  } else {
+    CeedInt num_input_fields, num_output_fields;
+    CeedOperatorField *input_fields, *output_fields;
+
+    ierr = CeedOperatorGetFields(op, &num_input_fields, &input_fields,
+                                 &num_output_fields, &output_fields); CeedChk(ierr);
+
+    CeedInt num_elem = 0;
+    ierr = CeedOperatorGetNumElements(op, &num_elem); CeedChk(ierr);
+    // Input FLOPs
+    for (CeedInt i = 0; i < num_input_fields; i++) {
+      if (input_fields[i]->vec == CEED_VECTOR_ACTIVE) {
+        CeedInt restr_flops, basis_flops;
+
+        ierr = CeedElemRestrictionGetFlopsEstimate(input_fields[i]->elem_restr,
+               CEED_NOTRANSPOSE, &restr_flops); CeedChk(ierr);
+        *flops += restr_flops;
+        ierr = CeedBasisGetFlopsEstimate(input_fields[i]->basis, CEED_NOTRANSPOSE,
+                                         op->qf->input_fields[i]->eval_mode, &basis_flops); CeedChk(ierr);
+        *flops += basis_flops * num_elem;
```

Comment:
Should the interface support the case where the same basis+field is used in `EVAL_INTERP` and `EVAL_GRAD`, so the marginal cost of `EVAL_INTERP` is zero?